### DISCUSSION TITLE
How to get the current value of a q-option-group

### DISCUSSION THREAD

**Discussion by lee3star** [quasarframework/quasar#12102] [General - Components / Directives / etc] [answered]
my code uses q-option-group to display 2 options: A and B. My intention is when the user chooses A, some components in the page are hidden, while the user chooses B, some other components in the page are hidden. 

here's my code(sorry cannot get codepen working): 

```
<template>
  <h3>Type</h3>
  <q-option-group
    v-model="type"
    :options="types"
    color="primary"
    inline
    @change="chooseTypechange"
    @mouseup="chooseTypemouseup"
    @mousedown="chooseTypemousedown"
  />
  <h3 v-if="()=>{this.type == `A`}">Only Show when A</h3>
  <h3 v-if="()=>{this.type == `B`}">Only Show when B</h3>
</template>

<script>
export default {
  data() {
    return {
      type: "A",
      types: [
        {
          label: "A",
          value: "A",
        },
        {
          label: "B",
          value: "B",
        },
      ],
    };
  },
  methods: {
    chooseTypechange() {
      console.log(`change: ${this.type}`);
    },
    chooseTypemouseup() {
      console.log(`mouseup: ${this.type}`);
    },
    chooseTypemousedown() {
      console.log(`mousedown: ${this.type}`);
    },
  },
};
</script>
```

Obviously, it's not working, in 3 ways: 

1st, doesn't matter whether I choose A or B, both "Only Show" H3 tags are shown
2nd, doesn't matter what I choose/change, chooseTypechange is never called
3rd, the weird part: 
- by default, option A is chosen
- if I click on A, from the console, I see: "mouseup: A" and "mousedown: A", which is fine
- then I click on B, from the console,  I see: "mouseup: A" and "mousedown: A". why A!? is this a bug?
- then I click on A again, from the console, I see: "mouseup: B" and "mousedown: B". why B now after I chose A?

**Comment by lee3star** [comment]
use error:

```
  <h3 v-if="()=>{this.type == `A`}">Only Show when A</h3>
  <h3 v-if="()=>{this.type == `B`}">Only Show when B</h3>
```

should be:

```
  <h3 v-if="this.type == `A`">Only Show when A</h3>
  <h3 v-if="this.type == `B`">Only Show when B</h3>
```

then it works as expected.

however, 2nd and 3rd problem remains. any ideas?

**Comment by MilosPaunovic** [comment, marked as answer]
[https://codepen.io/milospaunovic/pen/LYzopQb?editors=1011](https://codepen.io/milospaunovic/pen/LYzopQb?editors=1011)

**Reply by lee3star** [reply]
you are super, thanks a lot!
